specification/transport/udp/udp.tex

```diff
@@ -0,0 +1,337 @@
+    the variability is confined to the 23 least significant bits of the address only,
+    which is desirable because the IPv4 Ethernet MAC layer does not differentiate beyond the
+    23 least significant bits of the multicast group address.
+    That is, addresses that differ only in the 9 MSb collide at the MAC layer,
+    which is unacceptable in a real-time system; see RFC~1112 section 6.4.
+    Without this limitation, an engineer designing a network might inadvertently create a configuration
+    that causes MAC-layer collisions which may be difficult to detect.
+\end{remark}
+
+\begin{remark}
+    For example, the multicast group address for subject 42 is 239.0.0.42.
+    The multicast group address for a service transfer with the destination node-ID of 42 is 239.1.0.42.
+\end{remark}
+
+\begin{remark}
+    Per RFC~1112, in order to emit multicast traffic,
+    a limited level-1 implementation without the full support of IGMP and multicast-specific packet handling policies
+    is sufficient.
+    Thus, trivial nodes that are only required to publish messages on the network may be implemented
+    without the need for a full IGMP stack.
+
+    The reliance on IP multicasting exclusively allows baremetal implementations to omit ARP support.
+\end{remark}
+
+\begin{remark}
+    Due to the dynamic nature of the IGMP protocol,
+    a newly configured subscriber may not immediately receive data from the subject ---
+    a brief subscription initialization delay may occur
+    because the underlying IGMP stack needs to inform the router about its interest
+    in the specified multicast group by sending an IGMP membership report first.
+    Certain high-integrity applications may choose to rely on static switch configurations
+    to eliminate the subscription initialization delay.
+\end{remark}
+
+\subsection{UDP datagram payload format}
+
+The layout of the Cyphal/UDP datagram payload header is shown in the following snippet in DSDL notation
+(section~\ref{sec:dsdl}).
+The payload header is followed by the payload data, which is opaque to the protocol.
+
+\begin{samepage}
+\begin{minted}{python}
+# This 24-byte header can be aliased as a C structure with each field being naturally aligned:
+#
+#       uint8_t     version;
+#       uint8_t     priority;
+#       uint16_t    source_node_id;
+#       uint16_t    destination_node_id;
+#       uint16_t    data_specifier_snm;
+#       uint64_t    transfer_id;
+#       uint32_t    frame_index_eot;
+#       uint16_t    user_data;
+#       uint8_t[2]  header_crc16_big_endian;
+
+uint4 version
+# The version of the header format. This document specifies version 1.
+# Packets with an unknown version number must be ignored.
+
+void4
+
+uint3 priority
+# The values are assigned from 0 (HIGHEST priority) to 7 (LOWEST priority).
+# The numerical priority identifiers are chosen to be consistent with Cyphal/CAN.
+# Notice that this is the opposite of the priority ordering in the IP DSCP field.
```

Comment:
see comments above and revise accordingly.

Reply:
I understand that you are suggesting inverting the priority mapping here, such that higher values correspond to higher priority. I think retaining consistency with Cyphal/CAN and the existing experimental implementations of Cyphal/UDP is desirable, if only for the sake of consistency itself, so I suggest keeping this unchanged.